lib/rules/id-length.js

```diff
@@ -39,9 +41,9 @@ module.exports = function(context) {
         "VariableDeclarator": function(parent, node) {
             return parent.id === node;
         },
-        "Property": function(parent, node) {
+        "Property": properties ? function(parent, node) {
```

Comment:
I thought about it but somehow in my head the boolean was the opposite so it wouldn't have worked, I'll update!